src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1128,11 +1182,19 @@ CreateCitusTable(Oid relationId, char *distributionColumnName,
 	}
 	else if (distributionMethod == DISTRIBUTE_BY_NONE)
 	{
-		/*
-		 * This function does not expect to create Citus local table, so we blindly
-		 * create reference table when the method is DISTRIBUTE_BY_NONE.
-		 */
-		CreateReferenceTableShard(relationId);
+		if (replicationModel == REPLICATION_MODEL_2PC)
```

Comment:
Not sure about this, but I'm wondering whether this code becomes a bit easier to read if we give the table type as an argument. At this point, it's a bit weird that you can call this function for a hash-distributed table with REPLICATION_MODEL_2PC, or a null-shard-key table with REPLICATION_MODEL_COORDINATOR, and I'm not sure when it starts complaining.